### PR TITLE
feat:[CDS-83858]: helm chart for debezium-service

### PR DESCRIPTION
**Pull request by Aninditaa-ch**
## Describe your changes
Changes: 
Building the helm-chart for debezium-service, making it available for on-prem env. Previously it was on Saas only. All the files were added and made available with the required info added in config and values yaml files. 
Testing: 
Built and deployed successfully in devspaces, after testing by templating the chart. Service was working as expected.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56590)

### CODE REVIEW

File: debezium-service/charts/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# access-control
```
Comment (arya-harness) [inline debezium-service/charts/README.md:1]: install and run helm-docs to generate the readme, it currently contains access control

File: debezium-service/charts/templates/config.yaml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  POLL_INTERVAL_MS_SSCA: {{.Values.POLL_INTERVAL_MS_SSCA | quote}}
+  MONGODB_CONNECTION_STRING_SSCA: {{.Values.MONGODB_CONNECTION_STRING_SSCA | quote}}
+  JAVA_17_FLAGS: {{.Values.JAVA_17_FLAGS}}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{.Values.name}}
+stringData:
+  CF_CLIENT_API_KEY: {{.Values.CF_CLIENT_API_KEY}}
```
Comment (arya-harness) [inline debezium-service/charts/templates/config.yaml:120]: use external secrets and helper functions

File: debezium-service/charts/templates/config.yaml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+  {{- end }}
+data:
+  DEPLOY_MODE: {{.Values.DEPLOY_MODE}}
+  MEMORY: {{.Values.memory | quote}}
+  GOOGLE_APPLICATION_CREDENTIALS: /opt/harness/svc/cloud_stackdriver.json
+  STACK_DRIVER_LOGGING_ENABLED: {{.Values.STACK_DRIVER_LOGGING_ENABLED | quote}}
+  EVENTS_FRAMEWORK_REDIS_URL: {{.Values.EVENTS_FRAMEWORK_REDIS_URL}}
+  EVENTS_FRAMEWORK_USE_SENTINEL: {{.Values.EVENTS_FRAMEWORK_USE_SENTINEL | quote}}
+  EVENTS_FRAMEWORK_SENTINEL_MASTER_NAME: {{.Values.EVENTS_FRAMEWORK_SENTINEL_MASTER_NAME}}
+  EVENTS_FRAMEWORK_REDIS_SENTINELS: {{.Values.EVENTS_FRAMEWORK_REDIS_SENTINELS}}
```
Comment (arya-harness) [inline debezium-service/charts/templates/config.yaml:21]: Are all of these values required for debezium service?

**Review by arya-harness — CHANGES_REQUESTED** [review]

File: debezium-service/charts/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+            initialDelaySeconds: {{ .Values.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.readinessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.readinessProbe.successThreshold }}
+            failureThreshold: {{ .Values.readinessProbe.failureThreshold }}
+          {{- end }}
+          {{- if .Values.livenessProbe.enabled }}
+          livenessProbe:
+            httpGet:
+              path: /api/health
```
[the remark this answers is not in the document]
Comment (arya-harness) [reply debezium-service/charts/templates/deployment.yaml:109]: set those values in values.yaml

File: debezium-service/build/container/Dockerfile-debezium-service-multistage [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Copyright 2022 Harness Inc. All rights reserved.
```
Comment (arya-harness) [inline debezium-service/build/container/Dockerfile-debezium-service-multistage:1]: To be reviewed by @Vbhatia20

File: debezium-service/chart/templates/config.yaml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+#  EVENTS_FRAMEWORK_ENV_NAMESPACE: {{.Values.EVENTS_FRAMEWORK_ENV_NAMESPACE}}
+#  EVENTS_FRAMEWORK_REDIS_SSL_ENABLED: {{.Values.EVENTS_FRAMEWORK_REDIS_SSL_ENABLED | quote}}
+#  EVENTS_FRAMEWORK_REDIS_SSL_CA_TRUST_STORE_PATH: {{.Values.EVENTS_FRAMEWORK_REDIS_SSL_CA_TRUST_STORE_PATH}}
+#  EVENTS_FRAMEWORK_REDIS_SSL_CA_TRUST_STORE_PASSWORD: {{.Values.EVENTS_FRAMEWORK_REDIS_SSL_CA_TRUST_STORE_PASSWORD  | quote}}
+  REDIS_LOCK_CONFIG_REDIS_SENTINELS: {{ include "harnesscommon.dbconnectionv2.redisConnection" (dict "context" $) }}
+  REDIS_LOCK_CONFIG_SENTINEL_MASTER_NAME: harness-redis
+  REDIS_LOCK_CONFIG_USE_SENTINEL: {{ .Values.global.database.redis.installed | quote }}
+  REDIS_LOCK_CONFIG_REDIS_URL: {{ include "harnesscommon.dbconnectionv2.redisConnection" (dict "context" $) }}
+#  REDIS_LOCK_CONFIG_ENV_NAMESPACE: {{.Values.REDIS_LOCK_CONFIG_ENV_NAMESPACE | quote}}
+  IS_ENABLED_PIPELINE: {{.Values.global.IS_ENABLED_PIPELINE | quote}}
```
Comment (arya-harness) [inline debezium-service/chart/templates/config.yaml:30]: dont use .Values.global for ordinary configuration
global is reserved for configuration reused accross multiple charts.

use .Values.config instead

File: debezium-service/chart/templates/config.yaml [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  DEBEZIUM_NAME_PLG: {{.Values.global.DEBEZIUM_NAME_PLG | quote}}
+  MONGODB_NAME_PLG: {{.Values.global.MONGODB_NAME_PLG | quote}}
+  DATABASE_INCLUDE_LIST_PLG: {{.Values.global.DATABASE_INCLUDE_LIST_PLG | quote}}
+  REDIS_STREAM_SIZE_PLG: {{.Values.global.REDIS_STREAM_SIZE_PLG | quote}}
+  SNAPSHOT_MODE_PLG: {{.Values.global.SNAPSHOT_MODE_PLG | quote}}
+  MAX_QUEUE_SIZE_PLG: {{.Values.global.MAX_QUEUE_SIZE_PLG | quote}}
+  MAX_BATCH_SIZE_PLG: {{.Values.global.MAX_BATCH_SIZE_PLG | quote}}
+  MAX_QUEUE_SIZE_IN_BYTES_PLG: {{.Values.global.MAX_QUEUE_SIZE_IN_BYTES_PLG | quote}}
+  POLL_INTERVAL_MS_PLG: {{.Values.global.POLL_INTERVAL_MS_PLG | quote}}
+  MONGODB_CONNECTION_STRING_PLG: {{.Values.global.MONGODB_CONNECTION_STRING_PLG | quote}}
```
Comment (arya-harness) [inline debezium-service/chart/templates/config.yaml:83]: use helper function for mongo and move it to deployment.yaml
all database secrets and string are in deployment.yaml to be reffered directly from a secret or external secret

File: debezium-service/chart/templates/config.yaml [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+  IS_ENABLED_SSCA: {{.Values.global.IS_ENABLED_SSCA | quote}}
```
Comment (arya-harness) [inline debezium-service/chart/templates/config.yaml:85]: use .Values.global.ssca.enabled. There are charts which uses this flag

File: debezium-service/chart/templates/config.yaml [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+  MONGODB_SSL_ENABLED_SSCA: {{.Values.global.MONGODB_SSL_ENABLED_SSCA | quote}}
```
Comment (arya-harness) [inline debezium-service/chart/templates/config.yaml:86]: use mongo helper

File: debezium-service/chart/templates/config.yaml [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+  DEBEZIUM_NAME_SSCA: {{.Values.global.DEBEZIUM_NAME_SSCA | quote}}
+  MONGODB_NAME_SSCA: {{.Values.global.MONGODB_NAME_SSCA | quote}}
+  DATABASE_INCLUDE_LIST_SSCA: {{.Values.global.DATABASE_INCLUDE_LIST_SSCA | quote}}
+  REDIS_STREAM_SIZE_SSCA: {{.Values.global.REDIS_STREAM_SIZE_SSCA | quote}}
+  SNAPSHOT_MODE_SSCA: {{.Values.global.SNAPSHOT_MODE_SSCA | quote}}
+  MAX_QUEUE_SIZE_SSCA: {{.Values.global.MAX_QUEUE_SIZE_SSCA | quote}}
+  MAX_BATCH_SIZE_SSCA: {{.Values.global.MAX_BATCH_SIZE_SSCA | quote}}
+  MAX_QUEUE_SIZE_IN_BYTES_SSCA: {{.Values.global.MAX_QUEUE_SIZE_IN_BYTES_SSCA | quote}}
+  POLL_INTERVAL_MS_SSCA: {{.Values.global.POLL_INTERVAL_MS_SSCA | quote}}
+  MONGODB_CONNECTION_STRING_SSCA: {{.Values.global.MONGODB_CONNECTION_STRING_SSCA | quote}}
```
Comment (arya-harness) [inline debezium-service/chart/templates/config.yaml:97]: use mongo helper and move to deployment.yaml

File: debezium-service/chart/values.yaml [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+  name: "harness-default"
+readinessProbe:
+  httpGet:
+    path: /api/health
+    port: http-pms
+  timeoutSeconds: 2
+  periodSeconds: 5
+  failureThreshold: 5
+livenessProbe:
+  httpGet:
```
Comment (arya-harness) [inline debezium-service/chart/values.yaml:84]: use a enabled flag and set it to false, in ng-prod-manifests its not using livenessprobre

File: debezium-service/chart/templates/config.yaml [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+  MONGODB_NAME_APPLICATIONS: {{.Values.global.MONGODB_NAME_APPLICATIONS | quote}}
+  DATABASE_INCLUDE_LIST_APPLICATIONS: {{.Values.global.DATABASE_INCLUDE_LIST_APPLICATIONS | quote}}
+  REDIS_STREAM_SIZE_APPLICATIONS: {{.Values.global.REDIS_STREAM_SIZE_APPLICATIONS | quote}}
+  SNAPSHOT_MODE_APPLICATIONS: {{.Values.global.SNAPSHOT_MODE_APPLICATIONS | quote}}
+  MAX_QUEUE_SIZE_APPLICATIONS: {{.Values.global.MAX_QUEUE_SIZE_APPLICATIONS | quote}}
+  MAX_BATCH_SIZE_APPLICATIONS: {{.Values.global.MAX_BATCH_SIZE_APPLICATIONS | quote}}
+  MAX_QUEUE_SIZE_IN_BYTES_APPLICATIONS: {{.Values.global.MAX_QUEUE_SIZE_IN_BYTES_APPLICATIONS | quote}}
+  POLL_INTERVAL_MS_APPLICATIONS: {{.Values.global.POLL_INTERVAL_MS_APPLICATIONS | quote}}
+  FIELD_EXCLUDE_LIST_APPLICATIONS: {{.Values.global.FIELD_EXCLUDE_LIST_APPLICATIONS | quote}}
+  MONGODB_CONNECTION_STRING_APPLICATIONS: {{.Values.global.MONGODB_CONNECTION_STRING_APPLICATIONS | quote}}
```
Comment (arya-harness) [inline debezium-service/chart/templates/config.yaml:59]: check if this is a mongodb uri, and move to deployment.yaml if necessary

**Comment by Aninditaa-ch** [comment]
trigger uts

**Comment by Aninditaa-ch** [comment]
trigger uts

**Comment by Aninditaa-ch** [comment]
trigger messagecheck

**Comment by Aninditaa-ch** [comment]
trigger uts

**Comment by Aninditaa-ch** [comment]
trigger gitleaks

**Review by arya-harness — DISMISSED** [review]

**Review by arya-harness — APPROVED** [review]

**Review by jyotiraval — DISMISSED** [review]

**Comment by Aninditaa-ch** [comment]
trigger uts

**Comment by sahilhindwani-harness** [comment]
trigger gitleaks

**Comment by Aninditaa-ch** [comment]
trigger uts

**Review by arya-harness — APPROVED** [review]

**Review by jyotiraval — APPROVED** [review]

**Comment by Aninditaa-ch** [comment]
trigger uts

**Pull request merged**